cpp/src/rolling/rolling_detail.cuh

```diff
@@ -1212,6 +1213,54 @@ struct rolling_window_launcher {
                              stream,
                              mr);
   }
+
+  template <aggregation::Kind op, typename PrecedingIter, typename FollowingIter>
+  std::enable_if_t<(op == aggregation::COLLECT_LIST), std::unique_ptr<column>> operator()(
+    column_view const& input,
+    column_view const& default_outputs,
+    PrecedingIter preceding_begin_raw,
+    FollowingIter following_begin_raw,
+    size_type min_periods,
+    std::unique_ptr<aggregation> const& agg,
+    rmm::cuda_stream_view stream,
+    rmm::mr::device_memory_resource* mr)
+  {
+    return collect_list(input,
+                        default_outputs,
+                        preceding_begin_raw,
+                        following_begin_raw,
+                        min_periods,
+                        agg,
+                        stream,
+                        mr);
+  }
+
+  template <aggregation::Kind op, typename PrecedingIter, typename FollowingIter>
+  std::enable_if_t<(op == aggregation::COLLECT_SET), std::unique_ptr<column>> operator()(
+    column_view const& input,
+    column_view const& default_outputs,
+    PrecedingIter preceding_begin_raw,
+    FollowingIter following_begin_raw,
+    size_type min_periods,
+    std::unique_ptr<aggregation> const& agg,
+    rmm::cuda_stream_view stream,
+    rmm::mr::device_memory_resource* mr)
+  {
+    auto collect_result = collect_list(input,
+                                       default_outputs,
+                                       preceding_begin_raw,
+                                       following_begin_raw,
+                                       min_periods,
+                                       agg,
+                                       stream,
+                                       mr);
+
+    return lists::detail::drop_list_duplicates(lists_column_view(collect_result->view()),
+                                               null_equality::EQUAL,
+                                               nan_equality::ALL_EQUAL,
+                                               stream,
```

Comment:
In Spark, `NaNs` are treated as inequal.

Reply:
Fixed. Thx!